### DISCUSSION TITLE
How is day-to-day building and debugging facilitated?

### DISCUSSION THREAD

**Discussion by MaxJ345** [microsoft/msquic#2773] [Q&A] [answered]
I know it is [recommended](https://github.com/microsoft/msquic/blob/main/docs/BUILD.md#building-with-cmake) that the project be built using the PowerShell script - `./scripts/build.ps1` - but what do people use for day-to-day development? Do they use the CMake-related functionality built in to Visual Studio to build and debug the code?

Could someone please provide an example of how their day-to-day development workflow works for this project?

For context: I'm new to developing C/C++ projects in Visual Studio - and that includes using CMake.

**Comment by nibanks** [comment, marked as answer]
Personally, I do use PowerShell and it's associated scripts in my day to day work on MsQuic. I jump between platforms occasionally and don't want to have to remember any of the platform specific CMake options, so it's really nice that I can just run the same PS scripts everywhere.
